teps/0001-tekton-enhancement-proposal-process.md

```diff
@@ -0,0 +1,465 @@
+  - motivation for impactful user facing changes
+- ensure community participants are successfully able to drive changes
+  to completion across one or more releases while stakeholders are
+  adequately represented throughout the process
+
+This process is supported by a unit of work called a Tekton
+Enhancement Proposal (TEP). A TEP attempts to combine aspects of the
+following:
+
+- feature, and effort tracking document
+- a product requirements document
+- design document
+
+into one file which is created incrementally in collaboration with one
+or more [Working
+Groups](https://github.com/tektoncd/community/blob/master/working-groups.md)
+(WGs).
+
+This process does not block authors from doing early design docs using
+any means. It does not block authors from sharing those design docs
+with the community (during Working groups, on Slack, GitHub, …**.
+
+**This process acts as a requirement when a design docs is ready to be
+implemented or integrated in the `tektoncd` projects**. In other words,
+a change that impact other `tektoncd` projects or users cannot be
+merged if there is no `TEP` associated with it.
+
+## Motivation
+
+For cross project or new project proposal, an abstraction beyond a
+single GitHub issue seems to be required in order to understand and
+communicate upcoming changes to the Tekton community.
+
+In a blog post describing the [road to Go 2][], Russ Cox explains
+
+> that it is difficult but essential to describe the significance of a
+> problem in a way that someone working in a different environment can
+> understand
+
+As a project, it is vital to be able to track the chain of custody for
+a proposed enhancement from conception through implementation.
+
+Without a standardized mechanism for describing important
+enhancements, our talented technical writers and product managers
+struggle to weave a coherent narrative explaining why a particular
+release is important. Additionally, for critical infrastructure such
+as Tekton, adopters need a forward looking road map in order to plan
+their adoption strategy.
+
+Before this proposal, there is not a standard way or template to
+create project enhancements, but
+[suggestions](https://github.com/tektoncd/community/blob/master/process.md#proposing-features)
+on proposing a feature. We rely on documents hosted on Google docs,
+without a standard template explaining the change. Once a proposal is
+done, via a design docs, it tends to be hard to follow what happens
+with the proposal: updates on the proposal in reaction to comments,
+state of the proposal (when is it accepted, or rejected).
+
+The purpose of the TEP process is to reduce the amount of "tribal
+knowledge" in our community. By moving decisions from a smattering of
+mailing lists, video calls and hallway conversations into a well
+tracked artifact, this process aims to enhance communication and
+discoverability.
+
+A TEP is broken into sections which can be merged into source control
```

Comment:
i'm confused by what this means about sections which can be merged incrementally, could we link to the relevant section that explains that maybe?

Reply:
Right, I need to work on the template then :sweat_smile: In my head, part of the template **has to be** there for the initial merge of the proposal (while we keep discussing details during WGs), the rest can be completed later on (when we reached a decision).